frontend/public/components/utils/cog.tsx

```diff
@@ -127,4 +89,62 @@ export const ResourceCog = connectToModel(({actions, kindObj, resource, isDisabl
   />;
 });
 
+export class Cog extends DropdownMixin {
+  static factory: CogFactory = cogFactory;
+
+  onClick_(event, option) {
+    event.preventDefault();
+
+    if (option.callback) {
+      option.callback();
+    }
+
+    if (option.href) {
+      history.push(option.href);
+    }
+
+    this.hide();
+  }
+
+  render() {
+    const {options, anchor, isDisabled, id} = this.props;
+
+    return <div className={classNames('co-m-cog-wrapper', {'co-m-cog-wrapper--enabled': !isDisabled})} id={id}>
+      { isDisabled ?
+        <Tooltip content="disabled">
+          <div ref={this.dropdownElement} className={classNames('co-m-cog', `co-m-cog--anchor-${anchor || 'left'}`, {'co-m-cog--disabled' : isDisabled})} >
+            <span className={classNames('co-m-cog', 'co-m-cog__icon', 'fa', 'fa-cog', {'co-m-cog__icon--disabled' : isDisabled})}></span>
+          </div>
+        </Tooltip> :
+        <div ref={this.dropdownElement} onClick={this.toggle} className={classNames('co-m-cog', `co-m-cog--anchor-${anchor || 'left'}`, {'co-m-cog--disabled' : isDisabled})} >
+          <span className={classNames('co-m-cog', 'co-m-cog__icon', 'fa', 'fa-cog', {'co-m-cog__icon--disabled' : isDisabled})}></span>
+          { this.state.active && <CogItems options={options} onClick={(e, option) => this.onClick_(e, option)} /> }
```

Comment:
I thought arrow functions were discouraged for event handlers

https://reactjs.org/docs/faq-functions.html#arrow-function-in-render